openapi/src/main/java/io/helidon/openapi/OpenAPISupport.java

```diff
@@ -445,165 +323,208 @@ public static Builder builder() {
+         * Sets the location of the static OpenAPI document file.
          *
-         * @param operationID operation ID to which the server corresponds
-         * @param operationServer name of the server to add for this operation
+         * @param path non-null location of the static OpenAPI document file
          * @return updated builder instance
          */
-        public Builder addOperationServer(String operationID, String operationServer) {
-            Objects.requireNonNull(operationID, "operationID must be non-null");
-            Objects.requireNonNull(operationServer, "operationServer must be non-null");
-            apiConfigBuilder.addOperationServer(operationID, operationServer);
+        public Builder staticFile(String path) {
+            Objects.requireNonNull(path, "path to static file must be non-null");
+            staticFilePath = Optional.of(path);
             return this;
         }
 
-        /**
-         * Adds a path server for a given path.
-         *
-         * @param path path to which the server corresponds
-         * @param pathServer name of the server to add for this path
-         * @return updated builder instance
-         */
-        public Builder addPathServer(String path, String pathServer) {
-            Objects.requireNonNull(path, "path must be non-null");
-            Objects.requireNonNull(pathServer, "pathServer must be non-null");
-            apiConfigBuilder.addPathServer(path, pathServer);
-            return this;
+        private OpenApiStaticFile getExplicitStaticFile() {
+            Path path = Paths.get(staticFilePath.get());
+            final String specifiedFileType = typeFromPath(path);
+            final OpenAPIMediaTypes specifiedMediaType = OpenAPIMediaTypes.byFileType(specifiedFileType);
+
+            if (specifiedMediaType == null) {
+                throw new IllegalArgumentException("OpenAPI file path "
+                        + path.toAbsolutePath().toString()
+                        + " is not one of recognized types: "
+                        + OpenAPIMediaTypes.recognizedFileTypes());
+            }
+            final InputStream is;
+            try {
+                is = Files.newInputStream(path);
+            } catch (IOException ex) {
+                throw new IllegalArgumentException("OpenAPI file "
+                        + path.toAbsolutePath().toString()
+                        + " was specified but was not found");
+            }
+
+            try {
+                LOGGER.log(Level.FINE,
+                        () -> String.format(
+                                OPENAPI_EXPLICIT_STATIC_FILE_LOG_MESSAGE_FORMAT,
+                                path.toAbsolutePath().toString()));
+                return new OpenApiStaticFile(is, specifiedMediaType.format());
+            } catch (Throwable th) {
+                try {
+                    is.close();
+                } catch (IOException ex) {
+                    LOGGER.log(Level.FINE,
```

Comment:
Consider adding a suppressed throwable instead of logging.

Reply:
Yes, better. Fixed.